src/main/java/frc/robot/subsystems/drive/Gyroscope.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems.drive;
+
+import frc.robot.subsystems.drive.Drive;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.lang.Math;
+import com.analog.adis16470.frc.ADIS16470_IMU;
+import com.analog.adis16470.frc.ADIS16470_IMU.IMUAxis;
+
+public class Gyroscope extends SubsystemBase {
+
+    private Drive drive; 
+    private ADIS16470_IMU imu;
+
+    boolean drivingStraight = false;
+    double pValue;
+    double startTime;
+    double driftPerSecond;
+    
+    public Gyroscope(Drive drive){
+        this.imu = new ADIS16470_IMU();
+        this.drive = drive;
+        imu.setYawAxis(IMUAxis.kY);
+        gyroCalibrate();
+    }
+
+    public double getGyroAngle(){
+        double runTime = Timer.getFPGATimestamp() - startTime;
+        double drift = runTime * driftPerSecond;
+        return imu.getAngle() - drift;
+    }
+
+    public void resetGyro(){
+        imu.reset();
+        startTime = Timer.getFPGATimestamp();
+    }
+
+    public void driveStraightGyro(double forwardSpeed,double desiredAngle){ 
+        double error = (desiredAngle-getGyroAngle()) * .0014; 
+        drive.arcadeDrive(forwardSpeed, error);
+    }
+
+    public void backToStartingAngle(double speed){
+        double currentAngle = getGyroAngle();
+        double turnAngle = currentAngle % 360;
```

Comment:
doubles should be x.x to stop an implicit casting, ie 360.0

Reply:
Deleted back to starting angle method in gyro class, made it its own command, mapped it to the b button, and fixed the double formating

Reply:
in the two gyro commands